cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/kubernetes/IngressOperator.java

```diff
@@ -14,10 +14,19 @@
 import io.vertx.core.Future;
 import io.vertx.core.Vertx;
 
+import java.util.List;
+import java.util.function.Predicate;
+
 /**
  * Operations for {@code Ingress}es.
  */
 public class IngressOperator extends AbstractNamespacedResourceOperator<KubernetesClient, Ingress, IngressList, Resource<Ingress>> {
+    /**
+     * List of predicates that allows existing ingress annotations to be retained while reconciling the resources.
+     */
+    private static final List<Predicate<String>> INGRESS_ANNOTATION_IGNORELIST = List.of(
+            annotation -> annotation.startsWith("field.cattle.io")
```

Comment:
should we have a comment to make it clear that this annotation is coming from Rancher. I don't know Rancher so looking at the code I would not understand why this specific text is part of the ignore list.
A comment would be sufficient or we could have a well named constant to define it (i.e. something with `RANCHER_xxxxx`)